src/gc/gcinterface.h

```diff
@@ -463,6 +465,8 @@ class IGCHandleManager {
     virtual Object* GetDependentHandleSecondary(OBJECTHANDLE handle) = 0;
 
     virtual Object* InterlockedCompareExchangeObjectInHandle(OBJECTHANDLE handle, Object* object, Object* comparandObject) = 0;
+
+    virtual bool HandleAsyncPinnedHandles() = 0;
```

Comment:
This method enumerates all asyncpinned handles, and calls back the VM if the handle is pointing to overlapped that satisfies `overlapped->GetAppDomainId() != DefaultADID && overlapped->HasCompleted()`. So it basically a custom asyncpinned handle enumerator.

You will need to do something about the callback - it is the `overlapped->HandleAsyncPinHandle()` line in handletablecore.cpp. It may be best to pass the callback as a pointer+context, similar to how enumeration is done in other places on the GC/EE interface.

Once you do that, the method then can be called EnumerateCompletedAsyncPinnedHandlesOwnedByNonDefaultAppDomain. Or if you move the `overlapped->GetAppDomainId() != DefaultADID && overlapped->HasCompleted()` check into the callback as well, it can be just `EnumerateAsyncPinnedHandles`.

Also, it would be more appropriate for this method to be on IGCHandleStore because of it operates on the global GC handle store only. The VM can then call it as `GetGlobalHandleStore()->EnumerateAsyncPinnedHandles(...)`.